adaptive/learner/average_learner1D.py

```diff
@@ -0,0 +1,435 @@
+from copy import deepcopy
+from math import hypot
+
+import numpy as np
+import scipy.stats
+from sortedcollections import ItemSortedDict
+from sortedcontainers import SortedDict
+
+from adaptive.learner.learner1D import Learner1D, _get_intervals
+from adaptive.notebook_integration import ensure_holoviews
+
+
+class AverageLearner1D(Learner1D):
+    """Learns and predicts a noisy function 'f:ℝ → ℝ^N'.
+
+    New parameters (wrt Learner1D)
+    ------------------------------
+    delta : float
+        This parameter controls the resampling condition. A point is resampled
+        if its uncertainty is larger than delta times the smallest neighboring
+        interval.
+        We strongly recommend 0 < delta <= 1.
+    alpha : float (0 < alpha < 1)
+        The size of the interval of confidence of the estimate of the mean
+        is 1-2*alpha. We recommend to keep alpha=0.005.
+    neighbor_sampling : float (0 < neighbor_sampling <= 1)
+        Each new point is initially sampled at least a (neighbor_sampling*100)%
+        of the average number of samples of its neighbors.
+    min_samples : int (min_samples > 0)
+        Minimum number of samples at each point x. Each new point is initially
+        sampled at least min_samples times.
+    max_samples : int (min_samples < max_samples)
+        Maximum number of samples at each point x.
+    min_Delta_g : float (min_Delta_g >= 0)
```

Comment:
Why is this parameter called this way? It's not a very obvious name (from reading this doc-string at least).

Reply:
tl;dr: I agree, I changed the variable name and the doc-string [here](https://github.com/python-adaptive/adaptive/pull/283/commits/7550d1c64ae1d6d44c41b1d5774b5fba28c2d6cb). Do you think it is more clear now?

This variable was called min_Delta_g due to the original mathematical formulation of the problem. The true value of the function was called f(x), while the function with noise was g(x). The confidence intervals we build (the error bars) are for the mean value of g(x). Hence, I denoted the size of these intervals Δg. See [this notebook](https://nbviewer.jupyter.org/github/AlvaroGI/adaptive/blob/AverageLearner2D/test_strategy_5_new.ipynb) for a longer explanation with formulas.

Reply:
Looks good! Thanks for the explanation.